### PR TITLE
add a paramater --url-pattern to support drop package by match a url 

### PR DESCRIPTION
**Pull request by arthur657834**
### Describe what this PR does / why we need it
add a paramater --url-pattern to support drop package by match a url 

### Describe how to verify it
1. execute command  `chaos_dropnetwork --start --debug=true --remote-port 80 --url-pattern publish`
2. execute command  `curl http://www.qq.com/publish`, the result is `Empty reply from server`

### CODE REVIEW

File: exec/bin/dropnetwork/dropnetwork.go [hunk 1]
```diff
@@ -27,12 +27,14 @@ import (
...
 )
 
-var dropLocalPort, dropRemotePort string
+var dropLocalPort, dropRemotePort, dropUrlPattern string
 var dropNetStart, dropNetStop bool
 
 func main() {
 	flag.StringVar(&dropLocalPort, "local-port", "", "local port")
 	flag.StringVar(&dropRemotePort, "remote-port", "", "remote port")
+	fmt.Sprintf(`1234 dropRemotePort %s dropUrlPattern %s`, dropRemotePort, dropUrlPattern)
```
Comment (tiny-x) [inline exec/bin/dropnetwork/dropnetwork.go:36]: invalid fmt.Sprintf

**Review by tiny-x — APPROVED** [review]

File: exec/network_drop.go [hunk 2]
```diff
@@ -92,31 +96,40 @@ func (ne *NetworkDropExecutor) Exec(suid string, ctx context.Context, model *spe
...
-func (ne *NetworkDropExecutor) start(localPort, remotePort string, ctx context.Context) *spec.Response {
+func (ne *NetworkDropExecutor) start(localPort, remotePort, urlPattern string, ctx context.Context) *spec.Response {
 	args := fmt.Sprintf("--start --debug=%t", util.Debug)
 	if localPort != "" {
 		args = fmt.Sprintf("%s --local-port %s", args, localPort)
 	}
-	if remotePort != "" {
-		args = fmt.Sprintf("%s --remote-port %s", args, remotePort)
+	if urlPattern != "" {
+		args = fmt.Sprintf("%s --remote-port %s -m string --string %s --algo bm", args, remotePort, urlPattern)
```
Comment (tiny-x) [inline exec/network_drop.go:113]: It doesn't work here when I use the balde cli command, you can write it like this
````
if remotePort != "" {
		args = fmt.Sprintf("%s --remote-port %s", args, remotePort)
}
if urlPattern != "" {
		args = fmt.Sprintf("%s --url-pattern %s", args, urlPattern)
}
````
````
blade create network drop --remote-port 80 --url-pattern publish
````
{"code":604,"success":false,"error":"flag provided but not defined: -m\nUsage of /root/chaosblade/target/chaosblade-0.7.0/bin/chaos_dropnetwork:\n  -debug\n    \tset debug mode\n  -local-port string\n    \tlocal port\n  -remote-port string\n    \tremote port\n  -start\n    \tstart drop\n  -stop\n    \tstop drop\n  -url-pattern string\n    \turl pattern\n exit status 2"}

**Review by tiny-x — CHANGES_REQUESTED** [review]

File: exec/network_drop.go (hunk 2 above)
Comment (arthur657834) [reply exec/network_drop.go]: it does not work after modify as you said, is there something i missed？

File: exec/bin/dropnetwork/dropnetwork.go [hunk 3]
```diff
@@ -53,60 +54,75 @@ var cl = channel.NewLocalChannel()
...
 	}
-	handleDropSpecifyPort(remotePort, localPort, ctx)
+	handleDropSpecifyPort(remotePort, localPort, urlPattern, ctx)
 }
 
-func handleDropSpecifyPort(remotePort string, localPort string, ctx context.Context) {
+func handleDropSpecifyPort(remotePort string, localPort string, urlPattern string, ctx context.Context) {
 	var response *spec.Response
 	if localPort != "" {
 		response = cl.Run(ctx, "iptables",
```
Comment (tiny-x) [inline exec/bin/dropnetwork/dropnetwork.go:69]: Inbound traffic is not supported

**Review by tiny-x — APPROVED** [review]

**Comment by tiny-x** [comment]
#60

**Pull request merged**